File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/OnlineStore/Catigory/CatigoryForm.cs
using Serenity.ComponentModel;

namespace OnlineStoreAdminPanel.OnlineStore.Forms;

[FormScript("OnlineStore.Catigory")]
[BasedOnRow(typeof(CatigoryRow), CheckNames = true)]
public class CatigoryForm
{
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}

File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/ServerTypes/OnlineStore/CatigoryForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CatigoryForm {
    Name: StringEditor;
    Photo: StringEditor;
    Description: StringEditor;
}

export class CatigoryForm extends PrefixedContext {
    static readonly formKey = 'OnlineStore.Catigory';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatigoryForm.init)  {
            CatigoryForm.init = true;

            var w0 = StringEditor;

            initFormType(CatigoryForm, [
                'Name', w0,
                'Photo', w0,
                'Description', w0
            ]);
        }
    }
}